nhost/compose/config.go

```diff
@@ -360,17 +355,20 @@ func (c Config) functionsService() *types.ServiceConfig {
 		Restart:     types.RestartPolicyAlways,
 		Expose:      []string{"3000"},
 		Environment: c.functionsServiceEnvs().dockerServiceConfigEnv(),
-		HealthCheck: c.functionsServiceHealthcheck(time.Second*3, time.Minute*5),
+		HealthCheck: c.functionsServiceHealthcheck(time.Second*1, time.Minute*30), // 30 minutes is the maximum allowed time for a "functions" service to start, see more below
+		// Probe failure during that period will not be counted towards the maximum number of retries
+		// However, if a health check succeeds during the start period, the container is considered started and all
+		// consecutive failures will be counted towards the maximum number of retries.
 		Volumes: []types.ServiceVolumeConfig{
 			{
 				Type:   types.VolumeTypeBind,
 				Source: "..",
 				Target: "/opt/project",
 			},
 			{
-				Type:   types.VolumeTypeVolume,
-				Source: "functions_node_modules",
-				Target: "/opt/project/node_modules",
+				Type:   types.VolumeTypeBind,
+				Source: "../functions/node_modules",
+				Target: "/opt/project/functions/node_modules",
```

Comment:
Do we do this for both
`/opt/project/functions/node_modules` and
`/opt/project/node_modules`?

Reply:
We don't do for `/opt/project/node_modules`, but it's available for functions container anyway, this is how the data is persisted and shared with the host in the end

Reply:
I could probably omit this volume since "functions" container may create a folder. The root is shared anyway